### ISSUE TITLE
Wrong feature_ts when calculating batch features

### ISSUE THREAD

**Issue by artefactop** [GoogleCloudPlatform/fraudfinder#26]
Hi, 

I'm trying to follow this guide to set up Vertex in our organisation. I don't have much experience with this so I'm having some misunderstandings but I think I found a bug.

In https://github.com/GoogleCloudPlatform/fraudfinder/blob/main/02_feature_engineering_batch.ipynb the function `create_batch_features` is doing the following:

```python
# Create the query.
            query = query.replace("@END_DATE_TRAIN", f"'{date_query}'")
```
As it is replacing the `query` variable in the next iteration of the `for` there is nothing to replace because `@END_DATE_TRAIN` was already replaced.

so I change the code by:

```python
            # Create the query.
            query_ts = query.replace("@END_DATE_TRAIN", f"'{date_query}'")

            # Create the query job.
            job_config = bigquery.QueryJobConfig(
                destination=destination, write_disposition="WRITE_TRUNCATE"
            )
            print(query_ts)

            # Run the query to create daily feature table.
            job = bq_client.query(query_ts, job_config=job_config)
            _ = job.result()
```

I don't know if this is on purpose and the results should always has the same `feature_ts`
```sql
PARSE_TIMESTAMP('%Y-%m-%d %H:%M:%S', CONCAT(@END_DATE_TRAIN, ' ', STRING(TIME_TRUNC(CURRENT_TIME(), SECOND))), 'UTC') AS feature_ts,
```

Could you clarify it?

Thanks

**Comment by artefactop** [comment]
This is what I get in the customers table the `feature_ts` make none sense to me. Each row should have each own different `feature_ts` value.

![customers](https://user-images.githubusercontent.com/1199371/198060863-0966073d-ecc4-44f4-840b-6f9bee14f9fa.png)

**Comment by artefactop** [comment]
One more thing:
I changed this
```sql
PARSE_TIMESTAMP('%Y-%m-%d %H:%M:%S', CONCAT(@END_DATE_TRAIN, ' ', STRING(TIME_TRUNC(CURRENT_TIME(), SECOND))), 'UTC') AS feature_ts,
```
for this
```sql
a.TX_TS AS feature_ts,
```

and looks like now the features have the correct time

![customers_ts](https://user-images.githubusercontent.com/1199371/198065658-6fdbfecf-7c1e-485d-b74a-2c63e74da3b8.png)

**Comment by inardini** [comment]
Hi @artefactop,

Thanks for reaching out to the FraudFinder team. 

We want you to know that we recognized the issue related with the query variable and the date replacement within the backfill function. Also we verified the feature engineering process and we reproduced the incorrected behavior where each row of the entity tables have each row with the same feature_ts value.

Thanks to your feedback, we decided to simplify the feature engineering process which was originally designed to simulate a scheduled daily process. In the upcoming release, aggregated feature values will be computed based on transaction timestamp. Consequently, you will create new feature values in your feature store using the transaction timestamp. And a new batch of features will be retrieved by point-in-time query which will find the most up-to-date feature values looking up the transaction timestamp and the label timestamp. 

Best regards,

FraudFinder Team
